examples/simple_script/BUILD

```diff
@@ -17,9 +17,12 @@ ruby_binary(
 )
 
 ruby_test(
-    name = "default_test",
-    srcs = ["script_spec.rb"],
-    main = "script_spec.rb",
+    name = "script_spec",
```

Comment:
This is great, if I understand it correctly you are showing how to use `bazel test` command to run rspec with some nested files underneath? I've been having problem with it, and I think I know why.

On a typical Ruby/Rails project, spec files are layed out in a folder parallel to the libraries or app folder, which is not ideal for Bazel. But more than that, rspecs often use `require_relative` to load fixtures, and other files.

Here is a sample directory structure under `spec` folder of my gem [`sym`](https://github.com/kigster/sym):

```
spec
├── fixtures
│   ├── hamlet.enc
│   ├── hamlet.txt
│   ├── secrets.key
│   ├── secrets.yml
│   └── secrets.yml.enc
├── integration
│   └── sym_cli_spec.rb
├── spec_helper.rb
├── support
│   ├── aruba.rb
│   ├── contexts.rb
│   ├── fake_terminal.rb
│   └── shared_examples.rb
└── sym
    ├── app
    │   ├── args_spec.rb
    │   ├── cli_spec.rb
    │   ├── commands
    │   │   ├── encrypt_decrypt_spec.rb
    │   │   ├── generate_key_spec.rb
    │   │   ├── open_editor_spec.rb
    │   │   └── print_key_spec.rb
    │   ├── commands_spec.rb
    │   ├── input
    │   │   └── handler_spec.rb
    │   ├── keychain_spec.rb
    │   ├── password
    │   │   ├── cache_spec.rb
    │   │   └── providers_spec.rb
    │   └── private_key
    │       ├── detector_spec.rb
    │       └── handler_spec.rb
    ├── application_spec.rb
    ├── data_spec.rb
    ├── extensions
    │   └── instance_methods_spec.rb
    ├── magic_file_spec.rb
    └── sym_spec.rb
```

The `spec` folder itself is added to the `$LOAD_PATH` by the `rspec` gem command, so all files underneath can call `require 'spec_helper'` and have that work.

Beyond that, you might notice that there is a `support` folder with some critical files for reuse, and finally some pure data for testing encryption/decryption in the form of Shakespeare's Hamlet.  Specs rely on the fact that these files are located in a predictable location compared to the ruby files, and are able to read them when used via regular means, through bundler/rspec.

I noticed your code already puts in some thought about handing the data files, which is awesome.

### Specs and `LOAD_PATH`:

But let's get back to the second open PR on this repo, where I just am trying to get Circle to run rspec successfully without Bazel to start with. 

I wanted to try the following ideas:

 1. Have a single `BUILD` file at the top level of the gem instead of having many files deeply nested
 2. Be able to specify what is needed to run both binary and the tests. 
 3. The CLI is working fine. [But the rspec is not](https://github.com/bazelruby/ruby-monorepo/pull/3/files#diff-86b941af0ee0dfca3f9914a9121f4b73L28).

When `spec_helper.rb` file is being loaded, it currently *fails* trying to `require 'hello_world/cli'` which is the main file of the gem. So I added some [debugging statements here](https://github.com/bazelruby/ruby-monorepo/pull/3/files#diff-a85043f6a38b07ebd26f7980344d6c3cL19) so that I can see what ruby files surround the `spec_helper.rb` file. The result most likely does not surprise you, but for me it was very surprising :) 

This is the partial output of the Ruby files below the current directory from which tests are run. It's loading `ruby/gems/hello_world/lib/hello_world.rbruby/gems/hello_world/spec/spec_helper.rb` file, which is in a special path completely isolated by itself, and therefore it can't find the right files even they are all there.

What do you suggest with do to fix it? Should we add `ruby/gems/hello_world/lib` to `$LOAD_PATH`? Or should we change the folder where `spec_helper.rb` ends up to be closer to the original ruby layout?

```
ruby/gems/hello_world/spec/spec_helper.rb
ruby/gems/hello_world/spec/hello_world/cli_spec.rb
ruby/gems/hello_world/spec/hello_world/rake_spec.rb
ruby/gems/hello_world/lib/hello_world/parser.rb
ruby/gems/hello_world/lib/hello_world/cli.rb
ruby/gems/hello_world/lib/hello_world/version.rb
ruby/gems/hello_world/lib/hello_world.rbruby/gems/hello_world/spec/spec_helper.rb:27:in `require': cannot load such file -- hello_world/cli (LoadError)
	from ruby/gems/hello_world/spec/spec_helper.rb:27:in `<main>'

```